### PR TITLE
Workload ID: SPIFFE Federation documentation

### PR DESCRIPTION
**Pull request by strideynet**
Part of https://github.com/gravitational/teleport/issues/38927

More of a "reference"/"overview" page than a guide at this point. I'd like to revisit at somepoint in the future and write specific guides for federating with specific types of trust domain provider.

Do-not-merge until the actual releases have gone out and we can update the version number on this page.

### CODE REVIEW

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+different implementations to federate with one another. For example, a trust
+domain managed by Teleport Workload Identity could federate with a trust domain
+managed by SPIRE.
+
+The SPIFFE Federation process relies on the exchange of trust bundles between
+the trust domains. These trust bundles contain the certificates and public keys
+of the trust domain's issuer.
+
+Federation relationships are "one way" meaning that workloads within one trust
+domain can validate the identities of workloads in the other trust domain, but,
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:27]:
```suggestion
domain can validate the identities of workloads in the other trust domain, but
```

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+workloads in the other trust domain cannot validate the identities of workloads
+in the first trust domain. It is typical to therefore establish the federation
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:29]:
```suggestion
in the first trust domain. It is therefore typical to establish the federation
```

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+</Admonition>
+
+## Federation to Teleport Workload Identity
+
+This section discusses how to configure another trust domain to trust a Teleport
+Workload Identity hosted trust domain.
+
+In order to configure a trust domain to trust Teleport Workload Identity, the
+control plane of the trust domain must be able to access the SPIFFE Bundle
+Endpoint exposed on the Teleport Proxy.
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:44]:
```suggestion
Endpoint exposed on the Teleport Proxy Service.
```

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+The Teleport Workload Identity SPIFFE Bundle Endpoint implements the "https_web"
+profile as defined by the SPIFFE Federation specification. This means that the
+endpoint is served using the standard web TLS certificates configured for your
+Teleport Proxy.
+
+The SPIFFE Bundle Endpoint is exposed on the Teleport Proxy at the path
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:51]:
```suggestion
The SPIFFE Bundle Endpoint is exposed on the Teleport Proxy Service at the path
```

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+shape the future of Workload Identity and would love to
+[hear your feedback](mailto:product@goteleport.com).
+</Admonition>
+
+Federation is the ability for a trust relationship to be established between
+two trust domains, allowing workloads in one trust domain to validate the
+identities of workloads in the other trust domain.
+
+SPIFFE sets out a standard for federation that allows trust domains managed by
+different implementations to federate with one another. For example, a trust
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:18]: I'm not really sure from the introduction of this guide what functionality Teleport makes available. This sounds more like an overview of a SPIFFE concept. The only mention of Teleport is in this sentence, which sounds like a possible use case, but not a feature.

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+of Teleport Workload Identity.
+</Admonition>
+
+## Federation to Teleport Workload Identity
+
+This section discusses how to configure another trust domain to trust a Teleport
+Workload Identity hosted trust domain.
+
+In order to configure a trust domain to trust Teleport Workload Identity, the
+control plane of the trust domain must be able to access the SPIFFE Bundle
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:43]: What does "control plane" refer to in this context? Is control plane a well-understood concept in SPIFFE?

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+This section discusses how to configure a Teleport Workload Identity hosted
+trust domain to trust another trust domain.
+
+Teleport Workload Identity only supports federating with trust domains that
+offer a SPIFFE Bundle Endpoint that abides by the "https_web" profile as defined
+by the SPIFFE Federation specification.
+
+To configure Teleport Workload Identity to trust another trust domain, the
+`spiffe_federation` resource is used.
```
Comment (ptgott) [inline docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:64]: I think the passive voice makes this paragraph less clear. What uses the `spiffe_federation` resource? How does it use the resource?

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx (hunk 6 above)
Comment (strideynet) [reply docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:43]: It's fairly general engineering term, it's whatever component is managing/maintaining the other trust domain. Which depending on what software they're using, is maybe a specific thing like our "Auth Service" or maybe something totally different.

 It's hard to generalise here, it'll be easier to be more specific when we write up some guides for federating with specific implementations but I don't have time for that at the moment and with the current greenfield nature of Workload ID, we're working closely with most customers using it, so I'm not convinced writing a bunch of specific how-to guides is worth my time atm.

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx (hunk 5 above)
Comment (strideynet) [reply docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:18]: I had thought I'd captured this in the first sentence but I'll have another think at if I can make this clearer.

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx (hunk 7 above)
Comment (strideynet) [reply docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx]: 7a9be891d5

File: docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx (hunk 5 above)
Comment (strideynet) [reply docs/pages/enroll-resources/machine-id/workload-identity/federation.mdx:20]: [950cc23](https://github.com/gravitational/teleport/pull/46020/commits/950cc23ece2a6ff8b25f31306ba78588bc54a186)

**Review by ptgott — APPROVED** [review]

**Review by mmcallister — APPROVED** [review]

**Pull request merged**